tests/test_main.py

```diff
@@ -1,29 +1,31 @@
 import os
+import typing
 
 from click.testing import CliRunner
 
 import httpx
+from httpx._main import main
```

Comment:
What's prompted this change?

Can we continue to use `import httpx` and `httpx.main(...)` so that we're not importing private namespaces?

Reply:
Mypy then sees the `def main()` fallback defined in `httpx/__init__.py`, which is decidedly not a `click.BaseCommand` or subclass, so all the calls to `httpx.main()` then need a cast. Which is unfortunate.

I can see if we can't trick mypy by using a `TYPE_CHECKING` block that does the private import and leave the test using the normal, public import.

Reply:
> Without this, mypy sees the def main() fallback defined in httpx/__init__.py, which is decidedly not a click.BaseCommand or subclass, so all the calls to httpx.main() then need a typing.cast(click.BaseCommand, httpx.main). Which is unfortunate.

Ah, gotcha. You're referring to this...

https://github.com/encode/httpx/blob/57aa5b08a48c82a6e1a1500e131d94a31128b6f9/httpx/__init__.py#L47-L59

It looks to me like we should type the arguments on that.

```python
def main(*args: str) -> None:
    ...
```

Would that resolve the issue?

Reply:
Oh right, no it wouldn't.

Hrm. Perhaps we can put it inside a conditional `if not TYPE_CHECKING` block?

Reply:
Or else we just suck up the gnarly private import for now.

Perhaps that okay.

Reply:
I've put the import into a TYPE_CHECKING guard with a comment. The test is against the public import. This feels marginally better. Whadayathink?